ui/admin/app/routes/scopes/scope/session-recordings/index.js

```diff
@@ -4,5 +4,156 @@
+export default class ScopesScopeSessionRecordingsIndexRoute extends Route {
+  // =services
+  @service store;
+  @service router;
+  @service can;
+
+  // =attributes
+
+  queryParams = {
+    search: {
+      refreshModel: true,
+      replace: true,
+    },
+    times: {
+      refreshModel: true,
+      replace: true,
+    },
+    users: {
+      refreshModel: true,
+      replace: true,
+    },
+    scopes: {
+      refreshModel: true,
+      replace: true,
+    },
+    targets: {
+      refreshModel: true,
+      replace: true,
+    },
+    page: {
+      refreshModel: true,
+    },
+    pageSize: {
+      refreshModel: true,
+    },
+  };
+
+  allSessionRecordings;
+
+  /**
+   * Load all session recordings.
+   * @return {Promise<{ totalItems: number, sessionRecordings: [SessionRecordingModel], doSessionRecordingsExist: boolean, doStorageBucketsExist: boolean }>}
+   */
+  async model({ search, times, users, scopes, targets, page, pageSize }) {
+    const scope = this.modelFor('scopes.scope');
+    const { id: scope_id } = scope;
+    let sessionRecordings;
+    let totalItems = 0;
+    let doSessionRecordingsExist = false;
+    let doStorageBucketsExist = false;
+    const filters = {
+      created_time: [],
+      'create_time_values.user.id': [],
+      'create_time_values.target.scope.id': [],
+      'create_time_values.target.id': [],
+    };
+    times.forEach((time) => {
+      filters.created_time.push({ gte: new Date(time) });
+    });
+    users.forEach((user) => {
+      filters['create_time_values.user.id'].push({ equals: user });
+    });
+    scopes.forEach((scope) => {
+      filters['create_time_values.target.scope.id'].push({ equals: scope });
+    });
+    targets.forEach((target) => {
+      filters['create_time_values.target.id'].push({ equals: target });
+    });
+
+    if (
+      this.can.can('list scope', scope, {
+        collection: 'session-recordings',
+      })
+    ) {
+      const queryOptions = {
+        scope_id,
+        recursive: true,
+        query: { search, filters },
+        page,
+        pageSize,
+      };
+
+      sessionRecordings = await this.store.query(
+        'session-recording',
+        queryOptions,
+      );
+      totalItems = sessionRecordings.meta?.totalItems;
+      // Query all session reocordings for filtering values if entering route for the first time
+      if (!this.allSessionRecordings) {
+        await this.getAllSessionRecordings(scope_id);
+      }
+      doSessionRecordingsExist = Boolean(this.allSessionRecordings.length);
+      doStorageBucketsExist = await this.getDoStorageBucketsExist(scope_id);
+
+      return {
+        sessionRecordings,
+        doSessionRecordingsExist: doSessionRecordingsExist,
+        allSessionRecordings: this.allSessionRecordings,
+        totalItems,
+        doStorageBucketsExist: doStorageBucketsExist,
+      };
+    }
+  }
+
+  async getAllSessionRecordings(scope_id) {
```

Comment:
small nitpick: this method is missing a comment

Reply:
Fixed.